drivers/flash/flash_nxp_s32_qspi_nor.c

```diff
@@ -0,0 +1,1104 @@
+		QSPI_LUT_OP(QSPI_IP_LUT_INSTR_ADDR, QSPI_IP_LUT_PADS_1, 24U),
+		QSPI_LUT_OP(QSPI_IP_LUT_INSTR_WRITE, QSPI_IP_LUT_PADS_2, 8U),
+		QSPI_LUT_OP(QSPI_IP_LUT_INSTR_STOP, QSPI_IP_LUT_SEQ_END, QSPI_IP_LUT_SEQ_END),
+	},
+#endif
+
+#if QSPI_ANY_INST_HAS_PROP_EQ(writeoc, 2)
+	[QSPI_SEQ_PP_1_1_4] = {
+		QSPI_LUT_OP(QSPI_IP_LUT_INSTR_CMD, QSPI_IP_LUT_PADS_1, SPI_NOR_CMD_PP_1_1_4),
+		QSPI_LUT_OP(QSPI_IP_LUT_INSTR_ADDR, QSPI_IP_LUT_PADS_1, 24U),
+		QSPI_LUT_OP(QSPI_IP_LUT_INSTR_WRITE, QSPI_IP_LUT_PADS_4, 8U),
+		QSPI_LUT_OP(QSPI_IP_LUT_INSTR_STOP, QSPI_IP_LUT_SEQ_END, QSPI_IP_LUT_SEQ_END),
+	},
+#endif
+
+#if QSPI_ANY_INST_HAS_PROP_EQ(writeoc, 3)
+	[QSPI_SEQ_PP_1_4_4] = {
+		QSPI_LUT_OP(QSPI_IP_LUT_INSTR_CMD, QSPI_IP_LUT_PADS_1, SPI_NOR_CMD_PP_1_4_4),
+		QSPI_LUT_OP(QSPI_IP_LUT_INSTR_ADDR, QSPI_IP_LUT_PADS_4, 24U),
+		QSPI_LUT_OP(QSPI_IP_LUT_INSTR_WRITE, QSPI_IP_LUT_PADS_4, 16U),
+		QSPI_LUT_OP(QSPI_IP_LUT_INSTR_STOP, QSPI_IP_LUT_SEQ_END, QSPI_IP_LUT_SEQ_END),
+	},
+#endif
+};
+#endif /* !defined(CONFIG_FLASH_NXP_S32_QSPI_NOR_SFDP_RUNTIME) */
+
+static ALWAYS_INLINE Qspi_Ip_MemoryConfigType *get_memory_config(const struct device *dev)
+{
+#if defined(CONFIG_FLASH_NXP_S32_QSPI_NOR_SFDP_RUNTIME)
+	return &((struct nxp_s32_qspi_data *)dev->data)->memory_cfg;
+#else
+	return ((Qspi_Ip_MemoryConfigType *)
+		&((const struct nxp_s32_qspi_config *)dev->config)->memory_cfg);
+#endif
+}
+
+static ALWAYS_INLINE bool area_is_subregion(const struct device *dev, off_t offset, size_t size)
+{
+	Qspi_Ip_MemoryConfigType *memory_cfg = get_memory_config(dev);
+
+	return ((offset >= 0) && ((size + offset) <= memory_cfg->memSize));
+}
+
+static inline void nxp_s32_qspi_lock(const struct device *dev)
+{
+	struct nxp_s32_qspi_data *data = dev->data;
+
+#ifdef CONFIG_MULTITHREADING
+	k_sem_take(&data->sem, K_FOREVER);
+#else
+	ARG_UNUSED(data);
+#endif
+}
+
+static inline void nxp_s32_qspi_unlock(const struct device *dev)
+{
+	struct nxp_s32_qspi_data *data = dev->data;
+
+#ifdef CONFIG_MULTITHREADING
+	k_sem_give(&data->sem);
+#else
+	ARG_UNUSED(data);
+#endif
+}
+
+/* Must be called with lock */
+static int nxp_s32_qspi_wait_until_ready(const struct device *dev)
+{
+	struct nxp_s32_qspi_data *data = dev->data;
+	Qspi_Ip_StatusType status;
+	uint32_t timeout = 0xFFFFFF;
+	int ret = 0;
+
+	do {
+		status = Qspi_Ip_GetMemoryStatus(data->instance);
+		timeout--;
+	} while ((status == STATUS_QSPI_IP_BUSY) && (timeout > 0));
+
+	if (status != STATUS_QSPI_IP_SUCCESS) {
+		LOG_ERR("Failed to read memory status (%d)", status);
+		ret = -EIO;
+	} else if (timeout <= 0) {
+		LOG_ERR("Timeout, memory is busy");
+		ret = -ETIMEDOUT;
+	}
+
+	return ret;
+}
+
+#if !defined(CONFIG_FLASH_NXP_S32_QSPI_NOR_SFDP_RUNTIME)
+static int nxp_s32_qspi_read_status_register(const struct device *dev,
+					     uint8_t reg_num,
+					     uint8_t *val)
+{
+	struct nxp_s32_qspi_data *data = dev->data;
+	uint16_t lut_idx;
+	Qspi_Ip_StatusType status;
+	int ret = 0;
+
+	switch (reg_num) {
+	case 1U:
```

Comment:
Not maintainer, so not may job to complain, but wouldn't it be better to have name of these registers? Unless there are just magic numbers with secret names.

Reply:
`reg_num` is the Status Register number, so all of them are SR. I based on other drivers like https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/flash/nrf_qspi_nor.c#L506-L530